OpenSim/Common/Property.h

```diff
@@ -938,6 +952,88 @@ readSimplePropertyFromStream(std::istream& in)
        return SimTK::readUnformatted(in, values);
 }
 
+// We provide specializations for all types that contain floats to manage
+// the number of digits displayed. We rely on the default setting of how
+// `stringstream`s handle floats, where `precision` is the number of digits
+// printed, and the `stream` automatically determines whether float or
+// exponential notation is used.
+template <> inline void SimpleProperty<double>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << values[i];
+    }
+}
+
+template <> inline void SimpleProperty<SimTK::Vec3>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << "(" << values[i][0] << " ";
```

Comment:
Yes, it will. So, for a `OneValueProperty` of a `Vec3`, such as `mass_center`, we'll get something like `(0 0 0)` (and this falls in line with 3.3), but then if we have a property that is a list property of many `Vec3`s, we'll get something like `(0 0 0) (1 1 1) (2 2 2)`. Or at least that's the intention here.

Reply:
It seems to me that you would get `((0 0 0) (1 1 1) (2 2 2))`, unless I'm missing something.

Reply:
Oh oops, what you wrote is what I meant to write.

Reply:
Okay, as long as that's the expected behavior.